pxp/transaction_status.md

```diff
@@ -32,32 +38,94 @@ entry must comply with the following JSON schema:
-| stdout | string | output of the action, in case it did complete
-| stderr | string | possible errors of the action, in case it did complete
-| exitcode | integer | the exit code of the action, in case it did complete
+| transaction_id | string | the id of the queried transaction, in free text format
+| status | string | state of the action ("unknown", "success", "failure", "running", or "unknown")
+| output | object | contains the output of the requested action, if any
+| metadata | object | contains information about the execution of the requested action (mandatory in case *status* is not *unknown*)
+
+The following table provides more information about *status* values:
+
+| status value | description
+|--------------|------------
+| unknown | the agent has no information about the requested *transaction_id*
+| running | the action is still running
+| success | the action returned valid results on *stdout* (the returned *exitcode* is not taken into account)
+| failure | the agent failed to execute the action or received invalid results on *stdout*; a possible error will be reported in the *execution_error* entry
+| undetermined | the agent cannot determine whether or not the specified action completed (more details this should be reported in *execution_error*)
+
+The *metadata* object has analogous entries as the [RPC error message][2] one.
+The difference here is that *metadata* is optional; in fact the agent must
+not include such entry in case the requested transaction is *unknown*. When
+included, the only mandatory *action_metadata* entry is *start*. Nonetheless,
+the agent should include in the *status query* response as much information as
+it reliably can, about the requested transaction.
+
+Here is a description of *action_metadata* entries:
+
+| name | type | description
+|------|------|------------
+| execution_error | string | any error reported by the agent while executing the action or processing its output
+| start | string | this is the only mandatory entry - time entry in ISO8601 format indicating when the action started
+| end | string | time entry in ISO8601 format indicating when the action completed
+| stdout | string | any output of the action on stdout, in case it did complete
+| stderr | string | any output of the action on stderr
+| exitcode | int | exit code returned by the action
+
+Note that, for different *status* values, the agent should guarantee that the
+*stdout* entry includes:
+
+| status | stdout
+|--------|-------------
+| *unknown*, *running* | no *stdout* entry (the action must complete before the agent can guarantee that the results are in the expected format)
+| *failure*, *undetermined* | the action's output on stdout in a string format, without being validated
+| *success* | the action's output on stdout, in the action's specific format (the agent must have validated it)
+
+
+### Error Handling
+
+In case the agent cannot access the records related to specified transaction, it
```

Comment:
This seems implementation specific, I'd suggest rephrasing it to:
```
In case the agent is unable to determine the status of the specified transaction, it
...
```